packages/core/src/command/hl7-notification/hl7-notification-webhook-sender-direct.ts

```diff
@@ -62,69 +77,52 @@ export class Hl7NotificationWebhookSenderDirect implements Hl7NotificationWebhoo
       timestampString: sourceTimestamp,
     });
 
-    const bundle = prependPatientToBundle({
+    const newEncounterData = prependPatientToBundle({
       bundle: conversionResult,
       fhirPatient,
     });
     log(`Conversion complete and patient entry added`);
 
-    const nonSpecificUploadParams = {
-      s3Utils: this.s3Utils,
-      contentType: JSON_APP_MIME_TYPE,
-      log,
-      errorConfig: {
-        errorMessage: "Error uploading HL7 FHIR bundle to S3",
-        context: this.context,
-        captureParams: {
-          patientId,
-          cxId,
-          sourceTimestamp,
-        },
-        shouldCapture: true,
-      },
-    };
-
-    // TODO(lucas|ENG-257|2025-05-15): Remove this once new bucket flow is working correctly
-    // Upload to old bucket
-    const oldBundleFilename = buildHl7MessageFhirBundleFileKey({
+    await putAdtConversionBundle({
       cxId,
       patientId,
+      encounterId,
       timestamp: sourceTimestamp,
       messageId: getMessageUniqueIdentifier(message),
       messageCode,
       triggerEvent,
+      bundle: newEncounterData,
+      context: this.context,
+      s3Utils: this.s3Utils,
     });
-    await storeInS3WithRetries({
-      ...nonSpecificUploadParams,
-      bucketName: this.oldBucketName,
-      payload: JSON.stringify(bundle),
-      fileName: oldBundleFilename,
-    });
-    log(`Uploaded to S3 bucket: ${this.oldBucketName}. Filepath: ${oldBundleFilename}`);
 
-    // Upload to new bucket
-    const newBundleFilename = buildHl7MessageConversionFileKey({
+    const existingEncounterData = await getAdtSourcedEncounter({
       cxId,
       patientId,
-      timestamp: sourceTimestamp,
-      messageId: getMessageUniqueIdentifier(message),
-      messageCode,
-      triggerEvent,
+      encounterId,
     });
 
-    await storeInS3WithRetries({
-      ...nonSpecificUploadParams,
-      bucketName: this.bucketName,
-      payload: JSON.stringify(bundle),
-      fileName: newBundleFilename,
+    const currentEncounter = !existingEncounterData
+      ? newEncounterData
+      : mergeBundles({
+          cxId,
+          patientId,
+          existing: existingEncounterData,
+          current: newEncounterData,
+        });
+
+    const response = await putAdtSourcedEncounter({
+      cxId,
+      patientId,
+      encounterId,
+      bundle: currentEncounter,
     });
-    log(`Uploaded to S3 bucket: ${this.bucketName}. Filepath: ${newBundleFilename}`);
 
-    // TODO(lucas|ENG-257|2025-05-15): Generate latest.hl7.json and use it for presigned url
-    const oldBundlePresignedUrl = await this.s3Utils.getSignedUrl({
-      bucketName: this.oldBucketName,
-      fileName: oldBundleFilename,
+    const bundlePresignedUrl = await this.s3Utils.getSignedUrl({
+      bucketName: this.bucketName,
+      fileName: createFileKeyAdtLatest({ cxId, patientId, encounterId }),
```

Comment:
This is error prone. We're defining the name of the file to be downloaded here _in addition to_ `putAdtSourcedEncounter`. This exposes that function's behavior.

A better approach would be to return the filename as part of `putAdtSourcedEncounter`, and use it here. Same goes for the bucketName (which then doesn't even need to be on this class' constructor).

Reply:
addressed